desktop/cmp/persistenceManager/PersistenceManagerModel.ts

```diff
@@ -0,0 +1,345 @@
+
+    get value(): PlainObject {
+        return this.selectedObject?.value;
+    }
+
+    get objects(): PlainObject[] {
+        return this.views;
+    }
+
+    get selectedObject(): PlainObject {
+        return this.views.find(it => it.id === this.selectedId);
+    }
+
+    @computed
+    get canSave(): boolean {
+        const {value} = this;
+        if (!value) return false;
+        return (
+            this.isDirty && (this.canManageGlobal || value.isShared) && !this.loadModel.isPending
+        );
+    }
+
+    @computed
+    get isDirty(): boolean {
+        return !this.isEqualSkipAutosize(this.pendingValue, this.value);
+    }
+
+    get isShared(): boolean {
+        return !!this.selectedObject?.isShared;
+    }
+
+    // Internal persistence provider, used to save *this* model's state, i.e. selectedId
+    _provider;
+
+    constructor({
+        type,
+        noun,
+        onChangeAsync,
+        persistWith,
+        canManageGlobal,
+        enableTopLevelSaveButton = true,
+        newObjectFnAsync
+    }: PersistenceManagerProps) {
+        super();
+        makeObservable(this);
+
+        throwIf(
+            !isFunction(onChangeAsync),
+            'PersistenceManagerModel requires an `onChangeAsync` callback function'
+        );
+
+        this.type = type;
+        this.canManageGlobalFn = canManageGlobal;
+        this.enableTopLevelSaveButton = enableTopLevelSaveButton;
+        this.noun = noun || 'item';
+        this.newObjectFn = newObjectFnAsync ?? (() => ({}));
+        this.onChangeAsync = onChangeAsync;
+
+        // Set up internal PersistenceProvider -- fail gently
+        if (persistWith) {
+            try {
+                this._provider = PersistenceProvider.create({
+                    path: 'persistenceManager',
+                    ...persistWith
+                });
+
+                const state = this._provider.read();
+                if (state?.selectedId) this.selectedId = state.selectedId;
+                this.addReaction({
+                    track: () => this.selectedId,
+                    run: selectedId => this._provider.write({selectedId})
+                });
+            } catch (e) {
+                this.logError('Error applying persistWith', persistWith, e);
+                XH.safeDestroy(this._provider);
+                this._provider = null;
+            }
+        }
+
+        this.loadAsync();
+    }
+
+    // TODO - Carefully review if this method needs isStale checks, and how to properly implement them.
+    override async doLoadAsync(loadSpec: LoadSpec) {
+        const rawViews = await XH.jsonBlobService.listAsync({type: this.type, includeValue: true});
+
+        runInAction(() => (this.views = this.processRaw(rawViews)));
+
+        const {objects} = this;
+        // Auto-create an empty view if required
+        if (!objects.length) {
+            const newValue = await this.newObjectFn(),
+                newObject = await XH.jsonBlobService.createAsync({
```

Comment:
Wonder if we should be wrapping these in a `try` / `catch` and handling possible fetch exceptions